e2e-tests/contentful/cypress/integration/gatsby-image.js

```diff
@@ -0,0 +1,49 @@
+function hasSVGPlaceholder(el) {
+  el.children(`img`)
+    .should(`have.attr`, `src`)
+    .and(src => {
+      expect(src).to.match(/^data:image\/svg\+xml/)
+    })
+}
+
+function hasJPEGPlaceholder(el) {
+  el.children(`img`)
+    .should(`have.attr`, `src`)
+    .and(src => {
+      expect(src).to.match(/^data:image\/jpeg/)
+    })
+}
+
+function testGatsbyImage(type, testPlaceholder) {
+  if (testPlaceholder) {
+    cy.get(`[data-cy="${type}"]`)
+      .find(".gatsby-image-wrapper")
+      .each($el => {
+        const el = cy.wrap($el)
+        testPlaceholder(el)
+      })
+  }
+
+  cy.get(`[data-cy="${type}"]`).scrollIntoView()
+  // Wait for load
+  cy.wait(1000)
+
+  cy.get(`[data-cy="${type}"]`)
+    .find(".gatsby-image-wrapper > picture > img")
+    .each(($el, i) => {
+      cy.wrap($el).should(`have.attr`, `srcset`)
+      cy.wrap($el).should(`have.attr`, `src`)
+      cy.wrap($el).matchImageSnapshot(`${type}-${i}`)
+    })
+}
+
+describe(`gatsby-image`, () => {
+  beforeEach(() => {
+    cy.visit("/gatsby-image")
```

Comment:
We normally use `.waitForRouteChange()` after `.visit()`:

https://github.com/gatsbyjs/gatsby/blob/07ae70d27a997c4db1689163581a9d8034d91200/e2e-tests/development-runtime/cypress/integration/navigation/linking.js#L3

It's inside `gatsby-cypress`:

https://github.com/gatsbyjs/gatsby/blob/07ae70d27a997c4db1689163581a9d8034d91200/e2e-tests/development-runtime/cypress/support/index.js#L1

Reply:
added